### DISCUSSION TITLE
Assistance with what Im doing wrong

### DISCUSSION THREAD

**Discussion by choff5507** [hrynko/vue-pdf-embed#57] [Q&A] [answered]
I am trying to get my PDF to display in this plugin but I must be making some newbie mistakes. I believe Ive got the data formatted properly but the component never displays and I am not sure why. 

Im preparing the data like this:
```
pdfData = await new Blob([event.files[0]], {type: 'application/pdf'}).arrayBuffer()
pdfData = new Uint8Array(pdfData)
pdfData = `data:application/pdf;base64,<${pdfData}>`
```
At the end pdfData is what I am using for the **:source** . I then set the **v-if=displayPDF** to **True** but nothing displays. Im sure it's something small but could anyone point me to what Im doing wrong?

thank you

**Comment by hrynko** [comment, marked as answer]
Hi @choff5507,

I think you're missing the `target` of the `event`. Also you don't have to convert the input into base64, since the lib accepts `TypedArray` as a source as well. You can try the following:

```js
const file = event.target.files[0];
const reader = new FileReader();
reader.onload = () => {
  const typedarray = new Uint8Array(reader.result);
};
reader.readAsArrayBuffer(file);
```

Then use this `typedarray` as the source for you component. Hope this helps.

UPD: Added `readAsArrayBuffer` after noticing in thread.

**Reply by choff5507** [reply]
Thank you for the reply. I tried modifying the code you provided but it did not work. I noticed that there doesn't seem to be a target property on the event object. When I `console.log` I get `undefined`. Im not sure what Im missing but the documentation seems to be pretty straightforward. 

Let me describe what I am trying to accomplish and maybe with that there will be more insight. I have a page with several components (Im using PrimeVue). The user selects a PDF file to upload. Upon selection of this PDF file I have a method which gets this file off the `event` object and I want to convert to whatever format I need to in order to display the selected PDF file in the vue-pdf-embed component. I then set the `v-if` on the vue-pdf-embed component to `True` so it should display. Seems pretty straight forward but Ive got to be missing something. Any ideas?

Let me add also that I am using the script setup composition API. So perhaps Im missing something when assigning values to pdfData ? I tried adding the `.value` property on assignments but didn't seem to make a difference.

**Reply by hrynko** [reply]
So the problem is with getting the file, not with the conversion. Can you post an example of how you request a file and what value you get when you select it?

**Reply by choff5507** [reply]
Here's a link to what the event object returned by the upload component looks like. 
https://imgur.com/hmVKDZd
Hope this helps, this is what I am working with when a user selects a file. Thank you!

**Reply by hrynko** [reply]
Ok, so you don't need a `target`, but this file object is what you need. It should be convertible as described above. What kind of issue do you experience? Is it an issue of converting or passing the result as a prop?

**Reply by choff5507** [reply]
Here's what my code looks like now:
```
console.log(event.files[0])
    const file = event.files[0]
    const reader = new FileReader()
    reader.onload = () => {
        pdfData.value = new Uint8Array(reader.result)
    }
    displayPDF.value = true
```
Am I suppose to do something with the ```file``` variable? Seems to be defined but not used?
The error I get in the console is this:
**Expected Object | String | Uint8Array, got Undefined** 

Thanks!

**Reply by hrynko** [reply]
Indeed, the following line is missing:

```js
fileReader.readAsArrayBuffer(file);
```

You can put it after the `reader.onload` declaration.

**Reply by choff5507** [reply]
So, I tried the updated code, I couldn't get it to work. But I did manage to get it working. This is what worked for me:
```
const blob = new Blob([event.files[0]])
const objectUrl = URL.createObjectURL(blob)
pdfData.value = objectUrl
displayPDF.value = true
window.URL.revokeObjectURL(blob)
```
btw, is there an easy way to get the number of pages from the PDF Document proxy that is returned with ``loaded``` ? Any examples of zoom, page controls etc that you know of that I can implement with this plugin?

Thank you for all your help!

**Reply by hrynko** [reply]
There is a `numPages` property in PDFDocumentProxy. An example of page control can be found in the "Examples" section of the README. Zoom can be implemented with the `width`/`height` or `scale` props (will become available later this week).

**Reply by choff5507** [reply]
Ah ok thanks, looking forward to the update!

**Reply by choff5507** [reply]
I have one other question for you if you have some insight to it. Whenever I increment the page number on a PDF Document using this component it makes my page scroll to the top instead of keeping it in the same position as it was. The vue-pdf-embed component is a child component in another component. Ive tried several thngs to handle this but it still seems to do it. Do you have any advice on how to handle this ?

**Reply by hrynko** [reply]
There can be many reasons and solutions for this. Unfortunately, without knowing your code, I can hardly help you.

**Reply by choff5507** [reply]
Id be willing to share it with you if you're inclined. If not, I completely understand as well.
